src/sql/workbench/parts/dataExplorer/browser/connectionViewletPanel.ts

```diff
@@ -58,8 +58,9 @@ export class ConnectionViewletPanel extends ViewletPanel {
 			AddServerGroupAction.LABEL);
 		this._serverTreeView = this.instantiationService.createInstance(ServerTreeView);
 		this._activeConnectionsFilterAction = this._serverTreeView.activeConnectionsFilterAction;
-
-		this.objectExplorerService.registerServerTreeView(this._serverTreeView);
+		if (!this._serverTreeView) {
```

Comment:
>!this._serverTreeView [](start = 6, length = 21)

Is this correct? You're only registering it if it's undefined?

Reply:
Yes, register only if it doesn't exist. We only want to register the object explorer tree once.

Reply:
But you're creating it right above this with createInstance, that shouldn't normally returned undefined right?

---
In reply to: [276337654](https://github.com/Microsoft/azuredatastudio/pull/5076#discussion_r276337654) [](ancestors = 276337654)

Reply:
Updated to check from OE service.